scripts/sanity_chk/harness.py

```diff
@@ -121,6 +130,50 @@ def handle(self, line):
 
         self.process_test(line)
 
+class Pytest(Harness):
+    def handle(self, line):
+        cmd = [
+			'pytest',
+			'-s',
+			os.path.join(self.source_dir, self.pytest_root),
+			'--cmdopt',
+			self.running_dir,
+			'--junit-xml',
+			os.path.join(self.running_dir, 'report.html')
+        ]
+
+        with subprocess.Popen(cmd) as proc:
+            try:
+                proc.wait(15)
```

Comment:
Maybe we can use a `timeout` value defined in testcase/sample.yaml if given? I wonder if having a fixed time won't be a limitation for tests which might require more than 15s

Reply:
updated